cli/paas/wire_gen.go

```diff
@@ -51,3 +52,27 @@ func BuildApp(flags *pflag.FlagSet, configOverrides func(*config.Config)) (*Carr
 	return carrierClient, func() {
 	}, nil
 }
+
+// BuildInstallApp creates the Carrier Client for installation
+func BuildInstallApp(flags *pflag.FlagSet, configOverrides func(*config.Config)) (*CarrierClient, func(), error) {
+	configConfig, err := config.Load()
+	if err != nil {
+		return nil, nil, err
+	}
+	uiUI := ui.NewUI()
+	restConfig, err := config2.KubeConfig()
+	if err != nil {
+		return nil, nil, err
+	}
+	cluster, err := kubernetes.NewClusterFromClient(restConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+	carrierClient := &CarrierClient{
```

Comment:
I have a feeling we are doing it wrong. This Injector (#BuildInstallApp function), returns the same struct as BuildApp Injector. They both return a CarrierClient but in once case that struct has more fields filled in. From now on, when you see a CarrierClient instance in the code, you can't know which "version" it is.
I think wire's (and more generally, the dependency injection's) thinking is that you have an object that depends on others and all wire is doing is helping you avoid boilerplate code for just initializing the object. But the question here is:
Does a CarrierClient depend on a gitea.Client or not?
If yes then why is there a method creating a crippled version of CarrierClient (this one)?

Personally I think wire was probably created for way more complicated initializers than the ones we ever going to need but we were are to stick with wire, we should at least create different object types out of each initializer.

For example in this case, we could have BuildInstallApp returning actually an "InstallApp" or simply an "Installer". After all the Installer struct was created as the Object that knows how to install Carrier. Now it seems that knowledge is shared between the CarrierClient and the Installer object (one calls the other eventually).

Reply:
So, maybe an `InstallClient` which is the `CarrierClient` with the irrelevant fields gone ?

Reply:
AFAICT the `CarrierClient` is the object which encapsulates all the backend actions you can perform on a carrier installation. ... In some sense `Uninstall` would be a method of that, while `Install` is less so, as it is about bringing up an installation. so from that perspective splitting of an `InstallClient` would also make sense.

Part of the problem right now is I mixed both the re-org for install with the creation of the default org in a single PR.

Reply:
I think it should be like:

- Install cmd initializes an Installer (or `InstallCliet` if you prefer, but there should be no need for a separate `Installer` object)
- Install cmd calls Install on the above object
- Install cmd initializes a CarrierClient
- Install cmd calls CreatOrg on the above object

Reply:
I see, you are looking to merge the currently separate `Installer` structure of yours with a variant of the `CarrierClient` modified for install usage. Hm ... Should be doable. ... Heh, the addition of a `ui` field to `Installer` sort of goes in that direction already ... Add the `kubeClient` and we are there, I think. Maybe.

Reply:
Split of `CarrierClient` done with commit 50fd1cb6c863ec90f13d9059fbe6e27aeaa4d6d6.

No work has been done on merging `kubernetes.Installer` into the `InstallClient` yet.
While it would be the logical next step I am not fully convinced that it cannot be defered at this point.

Reply:
Ok. Continuing and looking further into the merge, flip/flopping. Fill the rest of the day.

Note: The `kubernetes.Installer` structure has some responsibilities which I believe are better left with it. Because they are things needed even before we have an `InstallClient`. I am talking about all the option handling. The only part of that we can move would be `ShowNeededOptions` ...

So ... Rename this to `kubernetes.DeploymentSet`, keep most of the option handling with it, move the `Show..` and `Install` methods over to the client. The `Show...` can become an internal function, even. That makes the `DS` mostly passive.

No UI facilities in the DeploymentSet. We do keep the global variable around, and hand that to the `Install` method as usual. Some renames in that area also for clarity.

Reply:
Done with commit 30b4fb7285551c378763a243cc36e963d03eceff.

:warning: Undone commit 5135955e4a0db387de755ea1bf73a4f8c85658b2 attaching an `ui` field to Deployment implementations, and `SetUI` interface.

The `ui` reference is in the same category as the `cluster` reference, not inherent information of the deployment, but an externally supplied thing for use by the current operation. We could go and aggregate these two references into a single structure. That would be the `InstallClient`, actually, pretty much. Doing that would also force us to move the core deployment related definitions out of `kubernetes` into `paas`, to avoid circular imports/dependencies. Not to be done today, if at all.